### PR TITLE
feat(api): Feature variable APIs now return default variable value when featureEnabled property is false.

### PR DESCRIPTION
**Pull request by mfahadahmed**
## Summary
- Feature variable APIs now return default variable value when featureEnabled property is false.

## Test plan
- Added unit tests.

### CODE REVIEW

File: packages/optimizely-sdk/lib/optimizely/index.js [hunk 1]
```diff
@@ -634,32 +634,25 @@ Optimizely.prototype._getFeatureVariableForType = function(featureKey, variableK
...
-        LOG_MESSAGES.USER_RECEIVED_VARIABLE_VALUE,
-        MODULE_NAME,
-        variableKey,
-        featureFlag.key,
-        variableValue,
-        userId
-      )
-    );
+    var value = projectConfig.getVariableValueForVariation(this.configObj, variable, decision.variation, this.logger);
+    if (value) {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/optimizely/index.js]: Should this be `value !== null`? What if `value` is an empty string?

File: packages/optimizely-sdk/lib/core/project_config/index.js [hunk 2]
```diff
@@ -519,7 +519,8 @@ module.exports = {
 
     var variableUsages = projectConfig.variationVariableUsageMap[variation.id];
     var variableUsage = variableUsages[variable.id];
-    return variableUsage ? variableUsage.value : variable.defaultValue;
+
+    return variableUsage ? variableUsage.value : null;
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/project_config/index.js]: The doc comment for this function is now incorrect, given this change.

**Comment by mfahadahmed** [comment]
build

**Comment by msohailhussain** [comment]
@jordangarcia are u going to review this PR?

**Comment by aliabbasrizvi** [comment]
@msohailhussain can you address conflict. I can merge right after.

**Comment by mfahadahmed** [comment]
@aliabbasrizvi Conflicts resolved. Please take a look again.

**Comment by jordangarcia** [comment]
@msohailhussain can you address conflicts

**Pull request merged**